opensrp-reveal/src/main/java/org/smartregister/reveal/widget/GeoWidgetFactory.java

```diff
@@ -63,20 +63,28 @@
     private JsonApi jsonApi;
 
     public static ValidationStatus validate(JsonFormFragmentView formFragmentView, RevealMapView mapView) {
+        ValidationStatus firstError = null;
         if (!Utils.isEmptyCollection(mapView.getValidators())) {
             for (METValidator validator : mapView.getValidators()) {
                 if (validator instanceof MinZoomValidator) {
                     MapboxMap mapboxMap = mapView.getMapboxMap();
                     double zoom = mapView.getMapboxMap().getCameraPosition().zoom;
                     if (mapboxMap != null && !validator.isValid(String.valueOf(zoom), false)) {
-
                         Toast.makeText(formFragmentView.getContext(), validator.getErrorMessage(), Toast.LENGTH_LONG).show();
-                        return new ValidationStatus(false, validator.getErrorMessage(), formFragmentView, mapView);
+                        ValidationStatus validationStatus = new ValidationStatus(false, validator.getErrorMessage(), formFragmentView, mapView);
+                        if (firstError == null) {
```

Comment:
This is should maybe be outside the `if (validator instanceof MinZoomValidator)` block?

Otherwise, there will only be the error from the MinZoomValidator, if any.

Reply:
I don't quite understand this concern, this is not not under any condition, its at the root of  the method validate

Reply:
Please counter-check the line in question, it's under another if-statement.

Reply:
I have removed this logic so that we validate one error at a time per widget